Plugins/Wox.Plugin.Calculator/Main.cs

```diff
@@ -38,33 +35,38 @@ public List<Result> Query(Query query)
 
             try
             {
-                var result = yampContext.Run(this._numberTranslator?.Translate(query.Search) ?? query.Search);
-                if (result.Output != null && !string.IsNullOrEmpty(result.Result))
+                var result = magesEngine.Interpret(query.Search);
+
+                if (result.ToString() == "NaN")
+                    result = context.API.GetTranslation("wox_plugin_calculator_not_a_number");
+
+                if (result is Function)
+                    result = context.API.GetTranslation("wox_plugin_calculator_expression_not_complete");
+
+
+                if (result != null && !string.IsNullOrEmpty(result.ToString()))
                 {
-                    string resultValue = this._numberTranslator?.TranslateBack(result.Result) ?? result.Result;
                     return new List<Result>
+                    { new Result
                     {
-                        new Result
+                        Title = result.ToString(),
+                        IcoPath = "Images/calculator.png",
+                        Score = 300,
+                        SubTitle = context.API.GetTranslation("wox_plugin_calculator_copy_number_to_clipboard"),
+                        Action = c =>
                         {
-                            Title = resultValue,
-                            IcoPath = "Images/calculator.png",
-                            Score = 300,
-                            SubTitle = "Copy this number to the clipboard",
-                            Action = c =>
+                            try
+                            {
+                                Clipboard.SetText(result.ToString());
+                                return true;
+                            }
+                            catch (ExternalException e)
                             {
-                                try
-                                {
-                                    Clipboard.SetText(resultValue);
-                                    return true;
-                                }
-                                catch (ExternalException e)
-                                {
-                                    MessageBox.Show("Copy failed, please try later");
-                                    return false;
-                                }
+                                MessageBox.Show("Copy failed, please try later");
+                                return false;
                             }
                         }
-                    };
+                    } };
```

Comment:
you may have to format this file to get a good code style.

Reply:
I will change that, thanks for pointing that out.